activerecord/lib/active_record/reflection.rb

```diff
@@ -759,6 +759,16 @@ def initialize(delegate_reflection)
         @delegate_reflection = delegate_reflection
         @klass = delegate_reflection.options[:anonymous_class]
         @source_reflection_name = delegate_reflection.options[:source]
+        if class_option = delegate_reflection.options[:class_name]
+          custom_message = class_option == source_reflection.class_name ?
+            "Remove :class_name option because it matches the source association class name" :
+            "Through association will always use the class of the source association specified.
+            Use a custom association with :class_name option as a through association :source to achive the same behavior."
```

Comment:
Spelling correction:
```suggestion
            Use a custom association with :class_name option as a through association :source to achieve the same behavior."
```